build/sync/README.md

```diff
@@ -0,0 +1,113 @@
+and/or directories that need to be kept in sync between the plugin repository and the template (this
+repo).
+
+For each set of paths, a set of actions to be performed is outlined. No more than one action of that set
+will be executed - the first one whose checks pass. Other actions are meant to act as fallbacks.
+The idea is to be able to e.g. overwrite a file if it has no local changes or apply a format-specific
+merge algorithm otherwise.
+
+Before running each action, the tool will check if any checks are defined for that action. If there
+are any, they will be executed and their results examined. If all checks pass, the action will be executed.
+If there is a check failure, the tool will locate the next applicable action according to the plan and
+start over with it.
+
+The synchronization plan can also run checks before running any actions, e.g. to check if the plugin and
+template worktrees are clean.
+
+Running
+-------
+
+The tool can be executed from the root of this repository with a command:
+```
+$ go run ./build/sync/main.go ./build/sync/plan.yml ../mattermost-plugin-github
+```
+
+(assuming `mattermost-plugin-github` is the plugin repository we want to synchronize with the template).
+
+plan.yml
+---------
+
+The `plan.yml` file (located in `build/sync/plan.yml`) consists of two parts:
+  - checks
+  - actions
+
+The `checks` section defines tests to run before executing the plan itself. Currently the only available such check is `repo_is_clean` defined as:
+```
+type: repo_is_clean
+params:
+  repo: template
+```
+The `repo` parameter takes one of two values:
+- `template` - the `mattermost-plugin-starter-template` repository
+- `plugin` - the repository of the plugin being updated.
+
+The `actions` section defines actions to be run as part of the synchronization.
+Each entry in this section has the form:
+```
+paths:
+  - path1
+  - path2
+actions:
+  - type: action_type
+    params:
+      action_parameter: value
+    conditions:
+      - type: check_type
+        params:
+          check_parameter: value
+```
+
+`paths` is a list of file or directory paths (relative to the root of the repository)
+synchronization should be performed on.
+
+Each action in the `actions` section is defined by its type. Currently supported action types are:
+  - `overwrite_file` - overwrite the specified file in the /plugin/ repository with the file in the /template/ repository.
+  - `overwrite_directory` - overwrite a directory.
+
+Both actions accept a parameter called `create` which determines if the file or directory should be created if it does not exist in the plugin repository.
+
+The `conditions` part of an action definition defines tests that need to pass for the
+action to be run. Available tests are:
+  - `exists`
+  - `file_unaltered`
+
+The `exists` check takes a single parameter - `repo` (referencing either the plugin or template repository) and it passes only if the file or directory the action is about to be run on exists.
```

Comment:
Why is a parameter needed? What is the use case for checking if a file exists in `template`? It it won't the check fail anyway?

Reply:
I left it availabe figuring that it may be useful. I've changed the exists checker to default to the `target` repo, so the parameter can be omitted.